flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java

```diff
@@ -1772,9 +1774,17 @@ private StateBackend loadStateBackend(ReadableConfig configuration, ClassLoader
                         + ContinuousFileMonitoringFunction.MIN_MONITORING_INTERVAL
                         + " ms.");
 
+        Configuration conf = (Configuration) getExecutionEnvironment().getConfiguration();
+        int readerParallelism = getParallelism();
+        if(readerParallelism == -1
+                && ClusterOptions.getSchedulerType(conf) == JobManagerOptions.SchedulerType.AdaptiveBatch) {
+            readerParallelism = conf.get(
```

Comment:
Maybe:
`if (readerParallelism == ExecutionConfig.PARALLELISM_DEFAULT && getConfig().isDynamicGraph()) {
       readerParallelism = getConfiguration().get(JobManagerOptions.ADAPTIVE_BATCH_SCHEDULER_DEFAULT_SOURCE_PARALLELISM);
}`

Reply:
-- if (readerParallelism == ExecutionConfig.PARALLELISM_DEFAULT && getConfig().isDynamicGraph())
yes, it's more concise.

--readerParallelism = getConfiguration().get(JobManagerOptions....);
actually, we cannot get config info via getConfiguration(), the result of getConfiguration() is empty.

Reply:
-> actually, we cannot get config info via getConfiguration(), the result of getConfiguration() is empty.
Why?

Reply:
I noticed that your original code logic was `getExecutionEnvironment().getConfiguration()`, I think this will lead to an empty configuration because the [StreamExecutionEnvironment#getExecutionEnvironment()](https://github.com/apache/flink/blob/master/flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java#:~:text=public%20static%20StreamExecutionEnvironment%20getExecutionEnvironment()%20%7B) will pass an empty configuration.

Reply:
[StreamExecutionEnvironment#getExecutionEnvironment()](https://github.com/apache/flink/blob/master/flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java#:~:text=public%20static%20StreamExecutionEnvironment%20getExecutionEnvironment()%20%7B) will load conf to initialize configuration, I have debugged the code, it will not lead to an empty configuration. On the contrary, getConfiguration() get the value of [configuration](https://github.com/apache/flink/blob/master/flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java#L208),  it is an empty configuration created in [StreamExecutionEnvironment()](https://github.com/apache/flink/blob/master/flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java#L221)